### PR TITLE
Translated config flow form and steps

### PR DESCRIPTION
**Pull request by springstan**
Issue affected: #3429

Description:
- added config flow form and steps translations to src/translations/en.json
- updated the config-flow files and show-dialog-config-flow.ts accordingly

Results:
add_new_area_prompt
alert_fail_adding_area
flow
submit
success

### CODE REVIEW

File: src/dialogs/config-flow/step-flow-abort.ts [hunk 1]
```diff
@@ -26,12 +26,20 @@ class StepFlowAbort extends LitElement {
...
+        )}
+      </h2>
       <div class="content">
         ${this.flowConfig.renderAbortDescription(this.hass, this.step)}
       </div>
       <div class="buttons">
-        <mwc-button @click="${this._flowDone}">Close</mwc-button>
+        <mwc-button @click="${this._flowDone}"
+          >${this.hass.localize(
+            "ui.panel.config.integrations.config_entry.close"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-abort.ts]:
```suggestion
            "ui.panel.config.integrations.config_flow.close"
```

File: src/dialogs/config-flow/step-flow-abort.ts [hunk 2]
```diff
@@ -26,12 +26,20 @@ class StepFlowAbort extends LitElement {
 
   protected render(): TemplateResult | void {
     return html`
-      <h2>Aborted</h2>
+      <h2>
+        ${this.hass.localize(
+          "ui.panel.config.integrations.config_entry.aborted"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-abort.ts]:
```suggestion
          "ui.panel.config.integrations.config_flow.aborted"
```

File: src/dialogs/config-flow/step-flow-create-entry.ts [hunk 3]
```diff
@@ -63,7 +63,9 @@ class StepFlowCreateEntry extends LitElement {
                           ${device.model} (${device.manufacturer})
                         </div>
                         <paper-dropdown-menu-light
-                          label="Area"
+                          label="${localize(
+                            "ui.panel.config.integrations.config_entry.area_picker_label"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-create-entry.ts]:
```suggestion
                            "ui.panel.config.integrations.config_flow.area_picker_label"
```

File: src/dialogs/config-flow/step-flow-create-entry.ts [hunk 4]
```diff
@@ -91,11 +93,19 @@ class StepFlowCreateEntry extends LitElement {
       <div class="buttons">
         ${this.devices.length > 0
           ? html`
-              <mwc-button @click="${this._addArea}">Add Area</mwc-button>
+              <mwc-button @click="${this._addArea}"
+                >${localize(
+                  "ui.panel.config.integrations.config_entry.add_area"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-create-entry.ts]:
```suggestion
                  "ui.panel.config.integrations.config_flow.add_area"
```

File: src/dialogs/config-flow/step-flow-create-entry.ts [hunk 5]
```diff
@@ -91,11 +93,19 @@ class StepFlowCreateEntry extends LitElement {
...
+                )}</mwc-button
+              >
             `
           : ""}
 
-        <mwc-button @click="${this._flowDone}">Finish</mwc-button>
+        <mwc-button @click="${this._flowDone}"
+          >${localize(
+            "ui.panel.config.integrations.config_entry.finish"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-create-entry.ts]:
```suggestion
            "ui.panel.config.integrations.config_flow.finish"
```

File: src/dialogs/config-flow/step-flow-create-entry.ts [hunk 6]
```diff
@@ -105,7 +115,11 @@ class StepFlowCreateEntry extends LitElement {
   }
 
   private async _addArea() {
-    const name = prompt("Name of the new area?");
+    const name = prompt(
+      this.hass.localize(
+        "ui.panel.config.integrations.config_entry.name_new_area"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-create-entry.ts]:
```suggestion
        "ui.panel.config.integrations.config_flow.name_new_area"
```

File: src/dialogs/config-flow/step-flow-create-entry.ts [hunk 7]
```diff
@@ -115,7 +129,11 @@ class StepFlowCreateEntry extends LitElement {
       });
       this.areas = [...this.areas, area];
     } catch (err) {
-      alert("Failed to create area.");
+      alert(
+        this.hass.localize(
+          "ui.panel.config.integrations.config_entry.failed_create_area"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-create-entry.ts]:
```suggestion
          "ui.panel.config.integrations.config_flow.failed_create_area"
```

File: src/dialogs/config-flow/step-flow-create-entry.ts [hunk 8]
```diff
@@ -134,7 +152,13 @@ class StepFlowCreateEntry extends LitElement {
         area_id: area,
       });
     } catch (err) {
-      alert(`Error saving area: ${err.message}`);
+      alert(
+        this.hass.localize(
+          "ui.panel.config.integrations.config_entry.error_saving_area",
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-create-entry.ts]:
```suggestion
          "ui.panel.config.integrations.config_flow.error_saving_area",
```

File: src/dialogs/config-flow/step-flow-form.ts [hunk 9]
```diff
@@ -87,14 +87,17 @@ class StepFlowForm extends LitElement {
                 <mwc-button
                   @click=${this._submitStep}
                   .disabled=${!allRequiredInfoFilledIn}
-                >
-                  Submit
+                  >${this.hass.localize(
+                    "ui.panel.config.integrations.config_entry.submit"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-form.ts]:
```suggestion
                    "ui.panel.config.integrations.config_flow.submit"
```

File: src/dialogs/config-flow/step-flow-form.ts [hunk 10]
```diff
@@ -87,14 +87,17 @@ class StepFlowForm extends LitElement {
...
+                  )}
                 </mwc-button>
 
                 ${!allRequiredInfoFilledIn
                   ? html`
-                      <paper-tooltip position="left">
-                        Not all required fields are filled in.
+                      <paper-tooltip position="left"
+                        >${this.hass.localize(
+                          "ui.panel.config.integrations.config_entry.not_all_fields_required"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-form.ts]:
```suggestion
                          "ui.panel.config.integrations.config_flow.not_all_fields_required"
```

File: src/dialogs/config-flow/show-dialog-config-flow.ts [hunk 11]
```diff
@@ -122,7 +122,13 @@ export const showConfigFlowDialog = (
               <ha-markdown allowsvg .content=${description}></ha-markdown>
             `
           : ""}
-        <p>Created config for ${step.title}.</p>
+        <p>
+          ${hass.localize(
+            "ui.panel.config.integrations.config_flow.external_step.created_config",
```
Comment (bramkragten) [inline src/dialogs/config-flow/show-dialog-config-flow.ts]:
```suggestion
            "ui.panel.config.integrations.config_flow.created_config",
```

File: src/dialogs/config-flow/step-flow-abort.ts (hunk 2 above)
Comment (springstan) [reply src/dialogs/config-flow/step-flow-abort.ts]: Resolved in https://github.com/home-assistant/home-assistant-polymer/pull/4140/commits/3cd31a26a03952acb61c9c8b02548fb0d3fda182

**Comment by springstan** [comment]
Fixed all suggestions with https://github.com/home-assistant/home-assistant-polymer/pull/4140/commits/3cd31a26a03952acb61c9c8b02548fb0d3fda182 :)

File: src/dialogs/config-flow/step-flow-form.ts [hunk 12]
```diff
@@ -87,14 +87,17 @@ class StepFlowForm extends LitElement {
...
+                  )}
                 </mwc-button>
 
                 ${!allRequiredInfoFilledIn
                   ? html`
-                      <paper-tooltip position="left">
-                        Not all required fields are filled in.
+                      <paper-tooltip position="left"
+                        >${this.hass.localize(
+                          "ui.panel.config.integrations.config_flow.not_all_fields_required"
```
Comment (bramkragten) [inline src/dialogs/config-flow/step-flow-form.ts]:
```suggestion
                          "ui.panel.config.integrations.config_flow.not_all_required_fields"
```

File: src/translations/en.json [hunk 13]
```diff
@@ -1215,6 +1215,17 @@
             "no_area": "No Area"
           },
           "config_flow": {
+            "aborted": "Aborted",
+            "close": "Close",
+            "finish": "Finish",
+            "submit": "Submit",
+            "not_all_fields_required": "Not all required fields are filled in.",
```
Comment (bramkragten) [inline src/translations/en.json]:
```suggestion
            "not_all_required_fields": "Not all required fields are filled in.",
```

**Review by bramkragten — APPROVED** [review]
OK to merge after comments

**Comment by springstan** [comment]
Renamed it with https://github.com/home-assistant/home-assistant-polymer/pull/4140/commits/c36e0926be88594547fccc2ea4d05e796cb91681 :)

**Pull request merged**